src/precompile.jl

```diff
@@ -1,7 +1,22 @@
 function _precompile_()
     ccall(:jl_generating_output, Cint, ()) == 1 || return nothing
-    precompile(Tuple{typeof(Parsers.parse), Type{Int64}, String})
-    precompile(Tuple{typeof(Parsers.parse), Type{Float64}, String})
-    precompile(Tuple{typeof(Parsers.parse), Type{Date}, String})
+
+    options = Parsers.Options()
+    pos = 1
+    val = "123"
+    len = length(val)
+    for T in (String, Int32, Int64, Float64, BigFloat, Dates.Date, Dates.DateTime, Bool)
```

Comment:
@rafaqz (and as fyi @nickrobinson251 ), no idea why, but including the `Dates.Date` and `Dates.DateTime` precompiles in Parsers.jl is causing https://github.com/JuliaData/CSV.jl/issues/981, i.e. precompiling CSV.jl on windows compleletly fails. Going to revert those two type precompiles for now and we can figure out what to do afterwards.

Reply:
can you only disable them for Windows?

Reply:
Its pretty weird that that happens. We could also rewrite a few of these methods to reduce compilation overheads, which may fix the problem.